huaweicloud/services/secmaster/resource_huaweicloud_secmaster_incident.go

```diff
@@ -0,0 +1,595 @@
+	createIncidentResp, err := createIncidentClient.Request("POST", createIncidentPath, &createIncidentOpt)
+	if err != nil {
+		return diag.Errorf("error creating Incident: %s", err)
+	}
+
+	createIncidentRespBody, err := utils.FlattenResponse(createIncidentResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.id", createIncidentRespBody)
+	if err != nil {
+		return diag.Errorf("error creating Incident: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceIncidentRead(ctx, d, meta)
+}
+
+func buildIncidentBodyParams(d *schema.ResourceData, cfg *config.Config) (map[string]interface{}, error) {
+	dataObject := map[string]interface{}{
+		"title":              utils.ValueIngoreEmpty(d.Get("name")),
+		"description":        utils.ValueIngoreEmpty(d.Get("description")),
+		"incident_type":      buildIncidentRequestBodyType(d.Get("type")),
+		"severity":           utils.ValueIngoreEmpty(d.Get("level")),
+		"handle_status":      utils.ValueIngoreEmpty(d.Get("status")),
+		"owner":              utils.ValueIngoreEmpty(d.Get("owner")),
+		"data_source":        buildIncidentRequestBodyDataSource(d.Get("data_source")),
+		"verification_state": utils.ValueIngoreEmpty(d.Get("verification_status")),
+		"ipdrr_phase":        utils.ValueIngoreEmpty(d.Get("stage")),
+		"simulation":         utils.ValueIngoreEmpty(d.Get("debugging_data")),
+		"labels":             utils.ValueIngoreEmpty(d.Get("labels")),
+		"close_reason":       utils.ValueIngoreEmpty(d.Get("close_reason")),
+		"close_comment":      utils.ValueIngoreEmpty(d.Get("close_comment")),
+		"environment":        buildEnvironmentOpts(d, cfg),
+		"domain_id":          cfg.DomainID,
+		"region_id":          cfg.GetRegion(d),
+	}
+
+	if v, ok := d.GetOk("first_occurrence_time"); ok {
+		firstOccurrenceTimeWithZ, err := formatInputTime(v.(string))
+		if err != nil {
+			return nil, err
+		}
+
+		dataObject["first_observed_time"] = firstOccurrenceTimeWithZ
+	}
+
+	if v, ok := d.GetOk("last_occurrence_time"); ok {
+		lastOccurrenceTimeWithZ, err := formatInputTime(v.(string))
+		if err != nil {
+			return nil, err
+		}
+
+		dataObject["last_observed_time"] = lastOccurrenceTimeWithZ
+	}
+
+	if v, ok := d.GetOk("planned_closure_time"); ok {
+		plannedClosureTimeWithZ, err := formatInputTime(v.(string))
+		if err != nil {
+			return nil, err
+		}
+
+		dataObject["sla"] = plannedClosureTimeWithZ
+	}
+
+	bodyParams := map[string]interface{}{
+		"data_object": dataObject,
+	}
+	return bodyParams, nil
+}
+
+func formatInputTime(timeFromSchema string) (string, error) {
+	inputTimeFormat := "2006-01-02T15:04:05.000-0700"
+	standardTime, err := time.Parse(standardTimeFormat, timeFromSchema)
+	if err != nil {
+		return "", fmt.Errorf("error parsing time to standard time: %s", err)
+	}
+
+	inputTimeWithoutZ := standardTime.Format(inputTimeFormat)
+	inputTimeWithZ := fmt.Sprintf(inputTimeWithoutZ[:23] + "Z" + inputTimeWithoutZ[23:])
```

Comment:
I think we can delete `fmt.Sprintf`, how about
```suggestion
	inputTimeWithZ := inputTimeWithoutZ[:23] + "Z" + inputTimeWithoutZ[23:]
```

Reply:
fixed